public/app/features/alerting/unified/components/rule-editor/PendingPeriodQuickPick.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+import { Button, Stack } from '@grafana/ui';
+
+import { formatPrometheusDuration, safeParsePrometheusDuration } from '../../utils/time';
+
+interface Props {
+  selectedPendingPeriod: string;
+  groupEvaluationInterval: string;
+  onSelect: (interval: string) => void;
+}
+
+export function PendingPeriodQuickPick({ selectedPendingPeriod, groupEvaluationInterval, onSelect }: Props) {
```

Comment:
Yea, certainly something we could make more abstract; I usually wait for at least a third repetition to avoid hasty abstractions and I don't imagine these components are going to be very "hot" (edited often) so I prefer to leave as-is for now.